models/transactions.go

```diff
@@ -0,0 +1,137 @@
+package models
+
+import (
+	"database/sql"
+	"log"
+	"time"
+)
+
+type Transaction struct {
+	ID    string `json:"id"`
+	Lines []*TransactionLine
+}
+
+type TransactionLine struct {
+	AccountID string `json:"account"`
+	Delta     int    `json:"delta"`
+}
+
+func (t *Transaction) IsValid() bool {
+	if len(t.Lines) == 0 {
```

Comment:
That's an interesting question. Technically it is a valid transaction, let's allow it for now. A no-op transaction is a possibility, especially with products that need to mark a certain sale as posted, but without any flow of money.